### DISCUSSION TITLE
Output results only provide the issues/vulnerabilities found  - Full list of  results including tests executed with no issues found is needed

### DISCUSSION THREAD

**Discussion by bhictor** [projectdiscovery/nuclei#6212] [Q&A]
**Feature request:**
We would like to have a way to indicate Nuclei that output files should contain all the execution results (not just the issues/findings). This would require also the addition of a new field for the test status in order to know the tests with issues and the ones without.
This feature would be specially useful in the json output, but I consider that for consistency should be a feature extensible for all the the output formats.
**Use case of this feature:**
A file containing all the tests executed it is needed because if there is no evidence of execution of a test we cannot control the test coverage and mantain a test execution log. In our particular project, we are interested in consuming the test execution outcome from Nuclei (for example the json output) and integrate automatically the results in a Test Management Tool like Xray.

Could you please consider this feature? We think it would be really useful for the Teams interested in using Nuclei integrated with Test Management Tools.
Thanks!

**Comment by GeorginaReeder** [comment]
Thanks so much for your feature request @bhictor , we'll take a look into it!

We also have a [Discord server](https://discord.com/invite/projectdiscovery), which you’re more than welcome to join. It's a great place to connect with fellow contributors and stay updated with the latest developments!

**Comment by ehsandeep** [comment]
@bhictor this is already supported in both format, cli and jsonl as well.

```
   -ms, -matcher-status  display match failure status
```

when `-ms` option is used, it shows both matched and failed matches.

**Comment by SchneiderSec** [comment]
@ehsandeep  You can use the -ms for cli output but it still doesn't output to the file. This functionality would be useful if wanting to retest and verify finding has been fixed programmatically. Is this possible?

**Comment by ehsandeep** [comment]
@SchneiderSec 

> You can use the -ms for cli output but it still doesn't output to the file.

what you mean, I can see the output file with failed result?

```console
$ echo https://example.com | nuclei -id tech-detect -ms -o output -silent
[tech-detect] [failed] [http] [info] example.com

$ cat output 
[tech-detect] [failed] [http] [info] example.com
```

**Comment by SchneiderSec** [comment]
@ehsandeep 
I'm talking about json and jsonl which would be much more useful from a programmatic standpoint as well as all the additional information that gets outputted to the json formats (response, template, tags, etc...). For example:

```cmd
nuclei -id tech-detect -ms -je test.json -silent -target https://example.com
[tech-detect] [failed] [http] [info] example.com

more test.json
[]
```

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
